lib/matplotlib/colorbar.py

```diff
@@ -488,6 +488,16 @@ def __init__(self, ax, mappable=None, *, cmap=None,
         if isinstance(mappable, contour.ContourSet) and not mappable.filled:
             self.add_lines(mappable)
 
+        # Link the Axes and Colorbar for interactive use
+        self.ax._colorbar = self
+        for x in ["_get_view", "_set_view", "_set_view_from_bbox",
+                  "drag_pan", "start_pan", "end_pan"]:
+            setattr(self.ax, x, getattr(self, x))
```

Comment:
Why are we defining the methods on the colorbar object, and then duplicating them on the axes?

Reply:
These methods need to have access to the Colorbar's norm data (inside of them I grab `self.norm.vmin`), so they don't have access to that data if they live only on the axes.

Reply:
OK, so what happens now if the user changes the xlim of the colorbar axis?  It looks to me that nothing happens....   Is that what we want?  

My concern here is that zoom on normal axes means "adjust the axes limits" (and populates a stack of previous limits).  Here it does that, but `set_xlim` will behave differently.

Reply:
If someone calls `set_xlim()` right away, that is respected. However, once someone starts interacting with the colorbar, the **norm** expands to the limits of the axis. Initially, the norm does not follow the xlim. That behavior could be argued either way I think, so I'm not inclined to change it in this PR... If someone wants to type in `set_xlim()` after having interacted that will be respected here too I think.

My preference is to keep the behavior as-is, and if someone has an issue with the mismatch between `set_xlim()` and `set_clim()` then we cross that bridge in a follow-up PR.

Here is a quick example to test with:
<details>
<summary>Click for demo code</summary>

```python
mport matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np
mpl.use('qt5agg')


rand_data = np.random.random((10, 10))*100

fig, [ax1, ax2, ax3] = plt.subplots(ncols=3)
cm = mpl.cm.get_cmap('plasma')
norm = mpl.colors.LogNorm(1, 100)
norm = mpl.colors.Normalize(1, 100)
sm = mpl.cm.ScalarMappable(norm=norm, cmap=cm)

mesh1 = ax1.imshow(rand_data, cmap=cm, norm=norm)
mesh2 = ax2.pcolorfast(rand_data, cmap=cm, norm=norm)
cb1 = fig.colorbar(sm, ax=[ax1, ax2],
                   extend='max', orientation='horizontal')
cb1.ax.set_xlim(0, 200)

pc = ax3.contourf(np.arange(1, 11), np.arange(1, 11), np.arange(100).reshape(10, 10),
                levels=np.arange(0, 100, 10)) #, vmin=0, vmax=50)
cb2 = fig.colorbar(pc, ax=ax3, orientation='vertical')


plt.show()
```
</details>

Reply:
Does this break `Axes.cla`?

Actually, I'm confused how these can work when they access `self.<Colorbar stuff>` when `self` should be an `Axes` that you've `setattr`d them on.

Reply:
The Axes methods are actually bound to the Colorbar object. Here is a print of `colorbar.ax._get_view` and `colorbar._get_view` showing they are both bound to the same object.

```python
<bound method Colorbar._get_view of <matplotlib.colorbar.Colorbar object at 0x7fc7b8eb4b10>>
<bound method Colorbar._get_view of <matplotlib.colorbar.Colorbar object at 0x7fc7b8eb4b10>>
```

> Does this break Axes.cla?

`cb.ax.cla()` does not remove these methods, so they are still there, but it does not impact anything other than interactivity, so I'm not sure how many people would be using `cb.ax.cla()` and expecting an original axes back and not doing a `cb.remove()` and creating a new one altogether... I'm not sure how we could replace the other methods, unless we kept them around somewhere just for that situation. I think this would be easier to fix properly if we had #20350 and could properly override these methods.

Reply:
It is definitely not very elegant, but I _think_ I got it to work as expected in the latest commit. Note that even with this addition of replacing the old axes interactivity, there is still an incorrect extend triangle hanging around, so calling `cla()` on a colorbar axes has probably never truly worked as desired.

Here is a quick script to test that with:
```python
import matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np

rand_data = np.random.random((10, 10))*100

fig, [ax1, ax2, ax3] = plt.subplots(ncols=3)
cm = mpl.cm.get_cmap('plasma')
norm = mpl.colors.LogNorm(1, 100)
norm = mpl.colors.Normalize(1, 100)
sm = mpl.cm.ScalarMappable(norm=norm, cmap=cm)

mesh1 = ax1.imshow(rand_data, cmap=cm, norm=norm)
mesh2 = ax2.pcolorfast(rand_data, cmap=cm, norm=norm)
cb1 = fig.colorbar(sm, ax=[ax1, ax2],
                   extend='max', orientation='horizontal')
# cb1.ax.cla()
# newcb = fig.colorbar(sm, cax=cb1.ax, extend='min', orientation='horizontal')

pc = ax3.contourf(np.arange(1, 11), np.arange(1, 11), np.arange(100).reshape(10, 10),
                levels=np.arange(0, 100, 10)) #, vmin=0, vmax=50)
cb2 = fig.colorbar(pc, ax=ax3, orientation='vertical')


plt.show()
```